scipy/special/cephes/bdtr.c

```diff
@@ -148,103 +148,92 @@
 
 #include "mconf.h"
 
-double bdtrc(k, n, p)
-int k, n;
-double p;
+double bdtrc(double k, int n, double p)
 {
     double dk, dn;
 
     if (npy_isnan(p)) {
```

Comment:
Now we will also need a NaN check for `k`.

Reply:
ok, I've got that changed, do we need a NaN check for `n` too? 
I'm assuming we are drop the if-check block in `scipy/special/_legacy.pxd` entirely, let me know if you think otherwise.